### PR TITLE
Propose: Restore list scroll position on navigation

### PR DESCRIPTION
**Pull request by pperotti**
Proposes a change to restore the movie list scroll position when users return from the details page, ensuring the previously viewed movie remains visible and highlighted.

## Change Summary
- **Feature**: restore-list-scroll-position
- **Problem**: List resets to top after returning from details, breaking UX
- **Solution**: Capture and restore scroll position using ViewModel state

## Artifacts
- [x] proposal.md - Problem statement and capabilities
- [x] design.md - Technical approach and decisions  
- [x] specs/list-scroll-position/spec.md - 7 requirements with 18 scenarios
- [x] tasks.md - 48 implementation tasks organized in 9 phases

Ready for review and implementation.

### CODE REVIEW

File: openspec/changes/restore-list-scroll-position/design.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+1. Add scroll position properties to existing MovieListViewModel
+2. Add RecyclerView scroll listener in MovieListFragment to capture position
+3. Intercept navigation events to save position before navigating away
+4. Restore position after data load (in data observer callback)
+5. Test back navigation, configuration changes, and edge cases
+6. No database migration or backwards compatibility concerns
+
+## Open Questions
+
+- Should scroll position be cleared when user manually refreshes the list (pull-to-refresh)?
```
Comment (pperotti) [inline openspec/changes/restore-list-scroll-position/design.md]: Yes

File: openspec/changes/restore-list-scroll-position/design.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+- Should this feature be configurable/toggleable by the user?
```
Comment (pperotti) [inline openspec/changes/restore-list-scroll-position/design.md]: Not needed

File: openspec/changes/restore-list-scroll-position/design.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+- Do we need to restore offset or only item index?
```
Comment (pperotti) [inline openspec/changes/restore-list-scroll-position/design.md]: You need to identify the offset position from the selectedMovieId property from the MainUiState.Success object.

**Review by pperotti — COMMENTED** [review]
Please address the answers to the open questions

**Review by pperotti — COMMENTED** [review]
LGTM. Let's move to the next phase
